packages/react-native-codegen/src/parsers/typescript/components/events.js

```diff
@@ -192,36 +194,49 @@ function getEventArgument(argumentProps, name: $FlowFixMe) {
   };
 }
 
+function findEvent(typeAnnotation: $FlowFixMe, optional: boolean) {
+  switch (typeAnnotation.type) {
+    // Check for T | null | undefined
+    case 'TSUnionType':
+      return findEvent(
+        typeAnnotation.types.filter(
+          t => t.type !== 'TSNullKeyword' && t.type !== 'TSUndefinedKeyword',
+        )[0],
+        optional ||
+          typeAnnotation.types.some(
+            t => t.type === 'TSNullKeyword' || t.type === 'TSUndefinedKeyword',
+          ),
+      );
+    // Check for (T)
+    case 'TSParenthesizedType':
+      return findEvent(typeAnnotation.typeAnnotation, optional);
+    case 'TSTypeReference':
+      if (
+        typeAnnotation.typeName.name !== 'BubblingEventHandler' &&
+        typeAnnotation.typeName.name !== 'DirectEventHandler'
```

Comment:
Can we use "positive" boolean clauses? I think they are easier to read and they remove the cognitive load of negating the clause:
```suggestion
  if(
        typeAnnotation.typeName.name === 'BubblingEventHandler' ||
        typeAnnotation.typeName.name === 'DirectEventHandler'
      ) {
        return {typeAnnotation, optional};
      }
```

nit: I would also `break;`, remove the `default` and `return null` at the end of the method.